[docs] Update and harmonize guids for deployment

marbre opened the pull request:
This updates and harmonizes the guides for the different deployment configurations. Several identical sections are provided as snippets.

* Updates docs regarding ROCm support
* Adds a subsection on downloading the runtime from a release
* Adds instructions on checking for CUDA and ROCm support

This superseds #18173 and #18655.

skip-ci: Doc updates only

marbre committed:
[docs] Update and harmonize guids for deployment

This updates and harmonizes the guides for the different deployment
configurations. Several identical sections are provided as snippets.

* Updates docs regarding ROCm support
* Adds a subsection on downloading the runtime from a release
* Adds instructions on checking for CUDA and ROCm support

This superseds #18173 and #18655.

skip-ci: Doc updates only

ScottTodd commented on docs/website/docs/guides/deployment-configurations/_iree-compiler-from-release.md:
> skip-ci: Doc updates only

This shouldn't be needed if you only make changes to the `docs/` directory, FYI.

That pattern should be matched here: https://github.com/iree-org/iree/blob/3ccd4f1a34776eb78f26dd3c2860212a27c3841e/build_tools/github_actions/configure_ci.py#L90-L91

which will set the list of enabled jobs to empty: https://github.com/iree-org/iree/blob/3ccd4f1a34776eb78f26dd3c2860212a27c3841e/build_tools/github_actions/configure_ci.py#L454-L459

which then causes these conditions to disable the jobs: https://github.com/iree-org/iree/blob/3ccd4f1a34776eb78f26dd3c2860212a27c3841e/.github/workflows/ci_linux_x64_clang.yml#L28-L30 https://github.com/iree-org/iree/blob/3ccd4f1a34776eb78f26dd3c2860212a27c3841e/.github/workflows/ci_linux_x64_clang_asan.yml#L28-L30
(any job that looks for `skip-ci` will also consider the paths modified)

marbre replied:
Thanks for all the pointers Scott!

ScottTodd commented on docs/website/docs/guides/deployment-configurations/_iree-run-module-driver-list.md:
Are these snippet files published to the website? We may want to exclude them somehow if so.

They would appear on https://iree.dev/sitemap.xml and possibly in search results.

Related: maybe these should go in a `snippets/` subfolder, either at the top level (i.e. `docs/website/docs/snippets/`) or local to where they are used (i.e. `docs/website/docs/guides/deployment-configurations/snippets/`)

marbre replied:
They are. As suggested I moved them into a local `snippets/` subfolder (I prefer to keep them close to where they are used) and excluded them via `exclude_docs`.

ScottTodd reviewed (approved):
Nice!

marbre committed:
Move snippets and exclude via `exclude_docs`

Moves the snippets to s `snippets` subfolder and adds those folders to
`exclude_docs` thus that the snippets get not included as separate files
in the site.
